arm-machinelearning/2016-05-01-preview/swagger/webservices.json

```diff
@@ -0,0 +1,915 @@
+          },
+          {
+            "in": "query",
+            "name": "$skiptoken",
+            "type": "string",
+            "description": "Continuation token for pagination.",
+            "required": false
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PaginatedWebServicesList"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.MachineLearning/webServices": {
+      "get": {
+        "tags": [
+          "WebServices"
+        ],
+        "operationId": "WebServices_GetWebServicesInSubscription",
+        "description": "Retrieve all Azure ML web services in the current Azure subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          },
+          {
+            "in": "query",
+            "name": "$skiptoken",
+            "type": "string",
+            "description": "Continuation token for pagination.",
+            "required": false
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PaginatedWebServicesList"
+            }
+          }
+        }
+      }
+    }    
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "description": "Azure subscription id.",
+      "in": "path",
+      "type": "string",
+      "required": true
+    },
+    "APIVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "type": "string",
+      "description": "The versiong of the Microsoft.MachineLearning resource provider API to be used.",
+      "required": true
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
+          "description": "Resource Id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource Name",
+          "type": "string"
+        },
+        "location": {
+          "description": "Resource Location",
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type",
+          "type": "string",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "WebService": {
+      "type": "object",
+      "description": "Instance of an Azure ML web service resource.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
It is upto you. After getting feedback from several customers about the frustration to create properties object while setting/creating things and looking things one level down while doing a get, we came up with this extension. Since all other RPs use this extension, it gives a consistent user experience to all the users across different services.

However, if you chose to not use the extension it is perfectly fine.